File: Tarea11 (RadixSort)/Tarea11.java
import java.util.Arrays;

public class Tarea11 {
    public static void countingSort(int[] arr, int exp) {
        int s = arr.length;
        int[] outputArray = new int[s];
        int[] countArray = new int[10];

        for (int j = 0; j < s; j++) {
            int idx = (arr[j] / exp) % 10;
            countArray[idx]++;
        }

        for (int j = 1; j < 10; j++)
            countArray[j] += countArray[j - 1];

        for (int j = s - 1; j >= 0; j--) {
            int idx = (arr[j] / exp) % 10;
            outputArray[countArray[idx] - 1] = arr[j];
            countArray[idx]--;
        }

        for (int j = 0; j < s; j++)
            arr[j] = outputArray[j];
    }

    public static void radixSort(int[] arr) {
        int max1 = Arrays.stream(arr).max().getAsInt();
        for (int exp = 1; max1 / exp > 0; exp *= 10)
            countingSort(arr, exp);
    }

    public static void main(String[] args) {
        int[] arr = {171, 46, 76, 91, 803, 25, 3, 67};
        System.out.println("Arreglo antes de ordenar: ");
        for (int n : arr) System.out.print(n + " ");

        radixSort(arr);

        System.out.println("\nDespués de ordenar el arreglo:");
        for (int n : arr) System.out.print(n + " ");
    }
}

File: Tarea11 (RadixSort)/Tarea11.py
def countingSort(arr, exp):
    s = len(arr)
    outputArray = [0] * (s)
    countArray = [0] * (10)
    for j in range(0, s):
        idx = arr[j] // exp
        countArray[idx % 10] += 1
    for j in range(1, 10):
        countArray[j] += countArray[j - 1]
    j = s - 1
    while j >= 0:
        idx = arr[j] // exp
        outputArray[countArray[idx % 10] - 1] = arr[j]
        countArray[idx % 10] -= 1
        j -= 1
    j = 0
    for j in range(0, len(arr)):
        arr[j] = outputArray[j]
def radixSort(arr):
    max1 = max(arr)
    exp = 1
    while max1 / exp > 1:
        countingSort(arr, exp)
        exp *= 10

arr = [171, 46, 76, 91, 803, 25, 3, 67]
print("Arreglo antes de ordenar: ")
for j in range(len(arr)):
    print("%d" % arr[j], end=" ")
radixSort(arr)
print("\nDespués de ordenar el arreglo: ")
for j in range(len(arr)):
    print("%d" % arr[j], end=" ")